typescript/sdk/src/middleware/liquidity-layer/LiquidityLayerRouterDeployer.ts

```diff
@@ -93,18 +93,26 @@ export class LiquidityLayerDeployer extends ProxiedRouterDeployer<
     this.logger(`Enroll CircleBridgeAdapters with each other`);
     // Hack to allow use of super.enrollRemoteRouters
     await super.enrollRemoteRouters(
-      objMap(contractsMap, (_, contracts) => ({
-        liquidityLayerRouter: contracts.circleBridgeAdapter,
-      })) as unknown as HyperlaneContractsMap<LiquidityLayerFactories>,
+      objMap(
+        // @ts-ignore Not sure how to specify the type guard
+        objFilter(contractsMap, (_, c) => !!c.circleBridgeAdapter),
+        (_, contracts) => ({
+          liquidityLayerRouter: contracts.circleBridgeAdapter,
+        }),
+      ) as unknown as HyperlaneContractsMap<LiquidityLayerFactories>,
       configMap,
     );
 
     this.logger(`Enroll PortalAdapters with each other`);
     // Hack to allow use of super.enrollRemoteRouters
     await super.enrollRemoteRouters(
-      objMap(contractsMap, (_, contracts) => ({
-        liquidityLayerRouter: contracts.portalAdapter,
-      })) as unknown as HyperlaneContractsMap<LiquidityLayerFactories>,
+      objMap(
+        // @ts-ignore Not sure how to specify the type guard
+        objFilter(contractsMap, (_, c) => !!c.portalAdapter),
+        (_, contracts) => ({
+          liquidityLayerRouter: contracts.portalAdapter,
+        }),
+      ) as unknown as HyperlaneContractsMap<LiquidityLayerFactories>,
```

Comment:
Maybe i'm not understanding the suggestion, but IIUC, these type gymnastics are necessary because of the `enrollRemoteRouters` function signature which expects a `HyperlaneContractsMap` parameterized by the `Factories` type of the deployer, so we have to convince the compiler that the generated map here "matches" that type?

Reply:
Yes but objFilter can do that for you

Reply:
Chatted offline to just punt since this code will get removed soon